components/director/pkg/operation/directive.go

```diff
@@ -0,0 +1,108 @@
+/*
+ * Copyright 2020 The Compass Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package operation
+
+import (
+	"context"
+	"fmt"
+
+	"github.com/kyma-incubator/compass/components/director/pkg/apperrors"
+	"github.com/kyma-incubator/compass/components/director/pkg/log"
+
+	gqlgen "github.com/99designs/gqlgen/graphql"
+	"github.com/kyma-incubator/compass/components/director/pkg/graphql"
+	"github.com/kyma-incubator/compass/components/director/pkg/persistence"
+)
+
+const ModeParam = "mode"
+
+type directive struct {
+	transact  persistence.Transactioner
+	scheduler Scheduler
+}
+
+// NewDirective creates a new handler struct responsible for the Async directive business logic
+func NewDirective(transact persistence.Transactioner, scheduler Scheduler) *directive {
+	return &directive{
+		transact:  transact,
+		scheduler: scheduler,
+	}
+}
+
+// HandleOperation enriches the request with an Operation information when the requesting mutation is annotated with the Async directive
+func (d *directive) HandleOperation(ctx context.Context, _ interface{}, next gqlgen.Resolver, op graphql.OperationType) (res interface{}, err error) {
+	resCtx := gqlgen.GetResolverContext(ctx)
+	mode, ok := resCtx.Args[ModeParam].(*graphql.OperationMode)
+	if !ok {
+		return nil, apperrors.NewInternalError(fmt.Sprintf("could not get %s parameter", ModeParam))
```

Comment:
I think this should be Internal Server error, because the our gql schema has specified a default OperationMode value. So in that sense if it's missing here, something has gone wrong on our side